spring-xd-spark-streaming-tests/src/test/java/org/springframework/xd/spark/streaming/AbstractSparkStreamingTests.java

```diff
@@ -116,6 +116,30 @@ public void testSparkProcessor() throws Exception {
 	}
 
 	@Test
+	public void testSparkTapStream() throws Exception {
+		HttpSource source = new HttpSource(shell);
+		String streamName =  testName.getMethodName() + new Random().nextInt();
+		String tapStreamName =  testName.getMethodName() + new Random().nextInt();
+		FileSink sink = new FileSink().binary(true);
+		try {
+			String stream = String.format("%s | counter", source);
+			createStream(streamName, stream);
+			String tapStream = String.format("tap:stream:"+ streamName + " > spark-word-count | %s --inputType=text/plain", sink);
+			createStream(tapStreamName, tapStream);
+			source.ensureReady().postData(TEST_LONG_MESSAGE);
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(StringContains.containsString("(foo,6)"))));
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(StringContains.containsString("(bar,5)"))));
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(StringContains.containsString("(test1,2)"))));
+			assertThat(sink, XDMatchers.eventually(XDMatchers.hasContentsThat(StringContains.containsString("(test2,1)"))));
+		}
+		finally {
+			streamOps.destroyStream(streamName);
```

Comment:
Since this class doesn't extend from `AbstractStreamIntegrationTest`, the stream deletion needs to be done explicitly.